codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/StructureGenerator.kt

```diff
@@ -285,6 +279,20 @@ class StructureGenerator(
                         .write("this.#L = #Q.invoke(block)", memberName, memberSymbol)
                         .closeBlock("}")
                 }
+
+                write("")
+
+                // render client side error correction function to set @required members
+                withBlock(
+                    "internal fun correctErrors(): Builder {",
```

Comment:
**Question**: We always call this immediately before `build()`. Are there any times we'd build a builder and _not_ call `correctErrors()`? If not, can we make this method private and invoke it directly from `build`?

Reply:
Yes, right now we only invoke this during deserialization (which is the error being corrected). It's unclear if we would default `@required` members or throw an error for input types.